### PR TITLE
Create tagging widget for Organigram

### LINKED ISSUE #1815: Create tagging widget for Organigram
## Acceptance Criteria
- The organigram should show a list of selected organigram items on a multi-select list

- [Figma Link](https://www.figma.com/proto/v9SGOBUqs5g5FSCjjwzIqR/DEEP---Wireframes?page-id=0%3A1&node-id=5020%3A8682&viewport=316%2C48%2C0.38&scaling=scale-down-width&starting-point-node-id=4051%3A442&show-proto-sidebar=1&hide-ui=1)

### PR DESCRIPTION
**Pull request by samshara**
- fixes  #1815

## Changes

* Create tagging widget for Organigram

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/components/OrganigramInput/styles.css [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    .card {
+        display: flex;
+        flex-grow: 1;
+        padding: var(--dui-spacing-medium);
+    }
+
+    .selected {
+        border: 1px solid var(--dui-color-accent);
```
Comment (AdityaKhatri) [inline app/components/OrganigramInput/styles.css]:
```suggestion
        border: var(--dui-width-separator-thin) solid var(--dui-color-accent);
```

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — COMMENTED** [review]
The imported library doesn't have a type definition or I couldn't find one.
Should we write some kind of type declaration for it or should we look for an alternative.
What do you think @samshara @tnagorra ?

File: app/components/OrganigramInput/styles.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    .disabled {
+        :global {
+            .org-tree-node-label {
+                pointer-events: none;
+            }
+        }
+    }
+}
+
+
```
Comment (tnagorra) [inline app/components/OrganigramInput/styles.css]:
```suggestion
```

File: app/components/OrganigramInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ), [value]);
+
+    if (!options) {
+        return null;
+    }
+
+    return (
+        <div
+            className={_cs(styles.container, className)}
+        >
```
Comment (tnagorra) [inline app/components/OrganigramInput/index.tsx]: Do we need a container div?

File: app/components/OrganigramInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            <OrgTree
+                data={options}
+                vertical
+                renderContent={(data: OrganigramDatum) => (
+                    <Card
+                        className={_cs(styles.card, isSelected(data) && styles.selected)}
+                    >
+                        { data.label }
+                    </Card>
+                )}
```
Comment (tnagorra) [inline app/components/OrganigramInput/index.tsx]: We should use useCallback for renderContent value

File: app/components/OrganigramInput/index.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        if (value?.some((v) => v === data.clientId)) {
+            onChange(value?.filter((v) => v !== data.clientId), name);
+        } else {
+            onChange(([] as string[]).concat(value ?? [], data.clientId), name);
+        }
+    }, [value, onChange, name]);
+
+    const isSelected = useCallback((d: OrganigramDatum) => (
+        value?.some((v) => v === d.clientId)
+    ), [value]);
```
Comment (tnagorra) [inline app/components/OrganigramInput/index.tsx]: Shouldn't this just be a `value?.find` ?

File: app/components/OrganigramInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        disabled,
+        readOnly,
+        options,
+    } = props;
+
+    const handleClick = useCallback((_: Event, data: OrganigramDatum) => {
+        if (value?.some((v) => v === data.clientId)) {
+            onChange(value?.filter((v) => v !== data.clientId), name);
+        } else {
+            onChange(([] as string[]).concat(value ?? [], data.clientId), name);
```
Comment (tnagorra) [inline app/components/OrganigramInput/index.tsx]: let's prefer this syntax `[...(value ?? []), data.clientId]`

File: app/components/OrganigramInput/index.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        name,
+        value,
+        onChange,
+        disabled,
+        readOnly,
+        options,
+    } = props;
+
+    const handleClick = useCallback((_: Event, data: OrganigramDatum) => {
+        if (value?.some((v) => v === data.clientId)) {
```
Comment (tnagorra) [inline app/components/OrganigramInput/index.tsx:47]: Can there be multiple data.clientId in `value`? Why are we using some instead of find?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/components/OrganigramInput/index.tsx (hunk 5 above)
Comment (samshara) [reply app/components/OrganigramInput/index.tsx]: boolean value needs to be returned so array.some is more semantic here.

File: app/components/OrganigramInput/index.tsx (hunk 7 above)
Comment (samshara) [reply app/components/OrganigramInput/index.tsx:47]: yes there will be multiple values in `value` props. The value props contains list of `clientId` right now. Maybe later we need to pass idSelector as there will be `id` passed from the server. Isn't `array.some` more semantic to check if item exists in an array and use it inside `if` condition.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Need to fix typing issues and write type declarations for the newly added library

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**